### PR TITLE
Searcher Controller has access to Executed Graph

### PR DESCRIPTION
**Pull request by mwu-tow**
### Pull Request Description
This PR adds access to the executed graph to the searcher controller. This will be used by a few different tasks, like #611 or #609. 

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Rust](https://github.com/luna/enso/blob/main/docs/style-guide/rust.md).
- [x] All code has been tested where possible.
- [ ] All code has been profiled where possible.

### CODE REVIEW

File: src/rust/ide/src/controller/searcher.rs [hunk 1]
```diff
@@ -8,7 +8,7 @@ use data::text::TextLocation;
 use enso_protocol::language_server;
 use flo_stream::Subscriber;
 use parser::Parser;
-
+use enso_protocol::language_server::MethodPointer;
```
Comment (farmaazon) [inline src/rust/ide/src/controller/searcher.rs:11]: import ordering

File: src/rust/ide/src/controller/searcher.rs [hunk 2]
```diff
@@ -438,20 +452,25 @@ mod test {
...
             let completion_response = language_server::response::Completion {
                 results: vec![1,5,9],
                 current_version: default(),
             };
+
             expect_call!(client.completion(
-                module      = "Test.Test".to_string(),
-                position    = TextLocation::at_document_begin().into(),
+                module      = data.module.path.file_path().clone(),
+                position    = TextLocation {line:0, column:5}.into(), // TODO should be something else likely
```
Comment (farmaazon) [inline src/rust/ide/src/controller/searcher.rs:473]: Indeed, this should be another position: end of "main = 2+2" or whatever the body is.

File: src/rust/ide/src/test.rs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    //! Data used to create mock IDE components.
+    //!
+    //! Contains a number of constants, so different parts of tests that mock different models use
+    //! consistent data.
+    #[allow(missing_docs)]
+    pub mod data {
+        pub const PROJECT_NAME:&str = "MockProject";
+        pub const MODULE_NAME:&str = "Mock_Module";
+        pub const CODE:&str = "main = 2 + 2";
+        pub const DEFINITION_NAME:&str = "main";
```
Comment (farmaazon) [inline src/rust/ide/src/test.rs:14]: Align

File: src/rust/ide/src/test.rs [hunk 4]
```diff
@@ -0,0 +1,28 @@
+//! Module for support code for writing tests.
+
```
Comment (farmaazon) [inline src/rust/ide/src/test.rs:2]: Add `#[cfg(test)]`

File: src/rust/ide/src/controller/searcher.rs [hunk 5]
```diff
@@ -342,16 +349,21 @@ impl Searcher {
...
-        let self_type   = None;
-        let position    = self.position.deref().into();
-        let request     = ls.completion(module,&position,&self_type,&return_type,&tags);
-        let data        = self.data.clone_ref();
-        let database    = self.database.clone_ref();
-        let logger      = self.logger.clone_ref();
-        let notifier    = self.notifier.clone_ref();
+    (&self, return_type:Option<String>, tags:Option<Vec<language_server::SuggestionEntryType>>)
+    -> FallibleResult<()> {
+        let ls             = &self.language_server;
```
Comment (wdanilo) [inline src/rust/ide/src/controller/searcher.rs:354]: lets use better name - ls -> lang_server :)

**Pull request merged**